Move moment-timezone away from initial webpack chunk

jsnajdr opened the pull request:
This PR attempts to remove `moment-timezone` library from the initial chunk and move it to the chunks that really need it.

The root cause of the issue is that `i18n-calypso` depends on `moment-timezone` and provides it to anyone who imports the `i18n` object or uses the `localize` HOC. But only very few callers actually use it. The solution in this PR is:
- activate the `i18n-calypso` package from the monorepo, so that we can change it
- make `i18n-calypso` import just `moment` instead of `moment-timezone`
- all places that use `moment.tz` have an explicit import from `moment-timezone`

The [result](http://iscalypsofastyet.com/push/29cfbb79034d062e319bc77d419797d7b392e80f) is nice:
```
chunk          stat_size           parsed_size           gzip_size
vendors~build  -730538 B (-24.2%)    -852695 B (-49.5%)   -14834 B (-5.7%)
```

@blowery what was your long-term plan with `i18n-calypso` and `moment`? Completely removing it from there? And relying 100% on `with-localized-moment`?

Many parts of my patch rely on the fact that we don't need to use the `moment` instance that `with-localized-moment` passes as prop:
```js
import moment from 'moment-timezone';

withLocalizedMoment( ( { moment: ignoredMoment, date, zone } ) => (
  <span>{ moment.tz( date, zone ).format( FORMAT ) }</span>
) );
```

The `moment` prop is ignored and instead we use the instance from top-level import. And everything still works 100% correctly. `moment` is a global singleton and the prop is always the same. The value provided by `withLocalizedMoment` is a `setState` call and rerender every time the locale changes.

jsnajdr commented:
By the way, (cc @blowery and @sirreal), activating the `i18n-calypso` package creates another duplication issue similar to #30378: this time it's `moment` that's bundled twice. And it will be harder to solve. The only solutions I know about are:
- pin exact versions in `packages/i18n-calypso/package.json`
- move to Yarn and its workspaces

sgomes reviewed (commented):
The approach in the PR seems reasonable, but won't it lead to incorrect results due to undefined locales? I could be getting it all wrong, but let me explain what I mean.

When importing from `moment-timezone`, you get a completely different `moment` object, separate from the one provided by `i18n-calypso`. This object won't have been initialised with a locale (`moment.locale(<locale>)`), like I believe `i18n-calypso` does. Doesn't this mean that all output from `moment-timezone` will use the default locale, after these changes?

blowery commented:
> When importing from moment-timezone, you get a completely different moment object, separate from the one provided by i18n-calypso.

moment-timezone should wrap the global moment instance; it doesn't clone it. Do you have a test showing that we drop the current locale when loading moment-timezone?

blowery commented:
> Does this mean that each of these chunks that need it will have its own copy of moment-timezone, or does it get put into some sort of common chunk by WebPack?

moment-timezone is big enough that it will likely get split off to a shared chunk

sgomes commented:
> > When importing from moment-timezone, you get a completely different moment object, separate from the one provided by i18n-calypso.
> 
> moment-timezone should wrap the global moment instance; it doesn't clone it. Do you have a test showing that we drop the current locale when loading moment-timezone?

Oh, I assumed that since these were all ES imports there would be no global namespace pollution. If moment does store something in the `window` object then `moment-timezone` can definitely reuse it.

blowery commented:
They're fake ES imports. :) Doesn't matter though.

moment-timezone is basically doing this:
```
import moment from 'moment';

moment.tz = // all the things

export default moment;
```

jsnajdr committed:
i18n-calypso: remove moment-timezone and move the dep to parent project

jsnajdr committed:
ClipboardButtonInput: use hooks

jsnajdr committed:
ProfileGravatar: doesn't need localize HOC

jsnajdr committed:
Me/ProfileLinks: remove unused props, state and CSS classes

Removes unused `showingForm` prop, `popoverPosition` state that has default (top) value and never changes,
and unused `popover-icon` CSS class.

jsnajdr committed:
i18n-calypso: move moment.js tests to separate file

jsnajdr committed:
i18n-calypso: remove moment-timezone tests

blowery commented:
@jsnajdr This all done now?

jsnajdr commented:
> This all done now?

Yes! Everything was merged except a few optional patches that I submitted today as separate PRs: #31724, #31725 and #31729.
